### ISSUE TITLE
Unable to generate source configs for supported source types: []

### ISSUE THREAD

**Issue by mrethers** [openshift/cluster-logging-operator#874]
**Describe the bug**
Creating a LogForwarding instance generates the following error:

level=warning msg="Unable to generate log confguraiton: Unable to generate source configs for supported source types: []"

This is regardless of whether I specify clusterlogging.openshift.io/logforwardingtechpreview: enabled in my cluster instance, or disableDefaultForwarding: true on my LogForwarding instance.

As soon as I delete the LogForwarding CR, everything gos back to working, but If I add it back, that message comes back, and fluentd is not being configured correctly.

**Environment**
Openshift 4.5.24 on IBM Cloud
Cluster-logging 4.5.0-202101160457.p0

```
apiVersion: logging.openshift.io/v1
kind: ClusterLogging
metadata:
  annotations:
    clusterlogging.openshift.io/logforwardingtechpreview: enabled
  name: instance
spec:
  collection:
    logs:
      fluentd: {}
      type: fluentd
  curation:
    curator:
      schedule: 30 3 * * *
    type: curator
  logStore:
    elasticsearch:
      nodeCount: 1
      redundancyPolicy: ZeroRedundancy
      storage: {}
    retentionPolicy:
      application:
        maxAge: 7d
    type: elasticsearch
  managementState: Managed
  visualization:
    kibana:
      replicas: 1
    type: kibana
```

```
apiVersion: logging.openshift.io/v1alpha1
kind: LogForwarding
metadata:
  namespace: openshift-logging
  name: instance
spec:
  outputs:
    - name: clo-default-output-es
      type: elasticsearch
      endpoint: 'elasticsearch.openshift-logging.svc:9200'
      secret:
        name: elasticsearch
  pipelines:
    - name: clo-default-app-pipeline
      inputSource: logs.app
      outputRefs:
        - clo-managaged-output-es
    - name: clo-default-infra-pipeline
      inputSource: logs.app
      outputRefs:
        - clo-managaged-output-es
```

**Logs**
cluster-logging operator logs when the LogForwarding CR is present:

```
time="2021-02-01T18:59:31Z" level=warning msg="Unable to generate log confguraiton: Unable to generate source configs for supported source types: []"
```

cluster-logging operator logs when the LogForwarding CR is **NOT** present
```
time="2021-02-01T18:59:31Z" level=info msg="Updating status of Fluentd"
time="2021-02-01T18:59:53Z" level=info msg="Collector container EnvVar change found, updating \"fluentd\""
time="2021-02-01T18:59:53Z" level=info msg="Collector volumes change found, updating \"fluentd\""
time="2021-02-01T18:59:53Z" level=info msg="Collector container volumemounts change found, updating \"fluentd\""
```

**Expected behavior**
Fluentd config should be generated from the default LogForwarding CR instance as indicated in the doc and using the provided sample CR. When this error happens, the fluentd config on the daemonset pods is empty.

**Actual behavior**
Error in the operator log and fluentd config in the daemonset pods present.

**To Reproduce**
1. Install Cluster Logging operator from OperatorHub
2.  Create an instance of cluster logging with Elasticsearch logStore
3. Add the logforwardingtechpreview annotation on the CR and create
4. Wait until fluentd pods start
5. Verify that the logs are flowing into Elasticsearch
6. Using the OML UI, create a default LogForwarding instance, not modifying anything from the provided sample
7. Wait until fluent pods restart
8. Check that fluentd config is empty (/etc/fluent/fluentd.conf)

**Additional context**

This just stopped working out-of-the-blue, I noticed the operator was upgraded last week and at first, fluentd was complaining about the elasticsearch certificate. I then proceeded to delete my existing instances (ClusterLogging and LogForwarding) and create another one. Logs started flowing again to the new ES instance, but as soon as I added my LogForwarding config back, I started running into that issue.

**Comment by jcantrill** [comment]
It does not look as though you defined a logforwarding CR to tell the CLO about the forwarding characteristics

**Comment by mrethers** [comment]
The issue comes up when I actually do create a LogForwarding CR. My LogForwarding CR is defined as follows:

```apiVersion: logging.openshift.io/v1alpha1
kind: LogForwarding
metadata:
  namespace: openshift-logging
  name: instance
spec:
  outputs:
    - name: clo-default-output-es
      type: elasticsearch
      endpoint: 'elasticsearch.openshift-logging.svc:9200'
      secret:
        name: elasticsearch
  pipelines:
    - name: clo-default-app-pipeline
      inputSource: logs.app
      outputRefs:
        - clo-managaged-output-es
    - name: clo-default-infra-pipeline
      inputSource: logs.app
      outputRefs:
        - clo-managaged-output-es```

**Comment by mrethers** [comment]
Took me a while but it turns out you cannot use the default output and pipeline names or it will fail with very little clue:

clo-default-output-es
clo-default-app-pipeline

That's what comes out of the sample CR when installing the operator from OLM. Would avoid confusion to change the sample CR to use different names.

**Comment by jcantrill** [comment]
likely fixed as part of 4.6 and GA
